rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionImpl.java

```diff
@@ -528,4 +533,32 @@ public BluetoothGattCharacteristic call(byte[] bytes) {
     public Observable<Integer> readRssi() {
         return rxBleRadio.queue(new RxBleRadioOperationReadRssi(gattCallback, bluetoothGatt, timeoutScheduler));
     }
+
+    @Override
+    public <T> Observable<T> queue(final RxBleRadioOperationCustom<T> operation) {
+        return rxBleRadio.queue(new RxBleRadioOperation<T>() {
+            @Override
+            protected void protectedRun() throws Throwable {
+                try {
+                    operation.execute(bluetoothGatt, gattCallback, rxBleRadio.scheduler())
```

Comment:
The operation is a public API, I think we should validate whether it behaves as we expect and check if returned value in not null.

Result of the assertion should clearly state the problem.